docs/CODE_GUIDELINES.md

```diff
@@ -198,164 +223,436 @@ exampleClass      ->InitExample();
   VALUE_X,
   VALUE_Y
 };
 ```
 
-### 11.4. Interfaces
-Use CamelCase for interface names and they have to be prefixed with an uppercase I. Filename has to match the interface name without the prefixed I, e.g. Logger.h
+### 8.4. Interfaces
+Use PascalCase and prefix with an uppercase I. Filename has to match the interface name without the prefixed I, e.g. Logger.h
 ```cpp
 class ILogger
 {
-  void Log(...) = 0;
+public:
+  virtual void Log(...) = 0;
+  virtual ~ILogger() {}
 }
 ```
 
-### 11.5. Classes
-Use CamelCase for class names and they have to be prefixed with an uppercase C. Filename has to match the class name without the prefixed C, e.g. Logger.cpp
+### 8.5. Classes
+Use PascalCase and prefix with an uppercase C. Filename has to match the class name without the prefixed C, e.g. Logger.cpp
 ```cpp
 class CLogger : public ILogger
 {
-  void Log(...)
+public:
+  void Log(...) override;
 }
 ```
 
-### 11.6. Methods
-Use CamelCase for method names and first letter has to be uppercase, even if the methods are private or protected.
+### 8.6. Methods
+Use PascalCase always, uppercasing the first letter even if the methods are private or protected.
 ```cpp
 void MyDummyClass::DoSomething();
 ```
 
-### 11.7. Variables
-Use CamelCase for variables. Type prefixing is discouraged.
+### 8.7. Variables
+Use CamelCase. Type prefixing (Systems Hungarian notation) is discouraged.
+
+#### Member variables
+Prefix nonstatic member variables with `m_`. Prefix static member variables with `ms_`.
+```cpp
+int m_variableA;
+static int ms_variableB;
+```
 
-#### Global Variables
-Prefix global variables with *g_*
+#### Global variables
+Prefix global variables with `g_`
 ```cpp
 int g_globalVariableA;
 ```
-**WARNING:** Avoid globals use. It increases the chances of submitted code to be rejected.
+**WARNING:** Avoid use of globals as far as reasonably possible. We generally do not want to introduce new global variables.
+
+**[back to top](#table-of-contents)**
 
-#### Member Variables
-Prefix member variables with *m_*
+## 9. Comments
+
+### 9.1. General
+
+Use `// ` for inline single-line and multi-line comments. Use `/* */` for the copyright comment at the beginning of the file. SPDX license headers are required for all code files (see example below).
+
+✅ Good:
 ```cpp
-int m_variableA;
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+// Nice comment
+
+// This can also continue for multiple lines:
+// I am the second line.
+```
+
+❌ Bad:
+```cpp
+/* A comment */
+//another comment
+```
+
+### 9.2. Doxygen
+
+New classes and functions are expected to have Doxygen comments describing their purpose, parameters, and behavior in the header file. However, do not describe trivialities - it only adds visual noise. Use the Qt style with exclamation mark (`/*! */`) and backslash for doxygen commands (e.g. `\brief`). 
```

Comment:
Instead of:

> However, do not describe trivialities - it only adds visual noise.

How about:

> Avoid trivialities - this is a sign that you should be more descriptive.

Reply:
Sure, your wording is fine.

Reply:
I get what you want to say though. If you find a wording that combines both, let me know...